nifi-commons/nifi-record-path/src/main/java/org/apache/nifi/record/path/paths/RecordPathCompiler.java

```diff
@@ -237,14 +240,16 @@ public static RecordPathSegment buildPath(final Tree tree, final RecordPathSegme
                         return new ReplaceNull(args[0], args[1], absolute);
                     }
                     case "concat": {
-                        final int numArgs = argumentListTree.getChildCount();
-
-                        final RecordPathSegment[] argPaths = new RecordPathSegment[numArgs];
-                        for (int i = 0; i < numArgs; i++) {
-                            argPaths[i] = buildPath(argumentListTree.getChild(i), null, absolute);
-                        }
-
-                        return new Concat(argPaths, absolute);
+                        return new Concat(getArgumentsForStringFunction(absolute, argumentListTree), absolute);
+                    }
+                    case "toLowerCase": {
+                        return new ToLowerCase(getArgumentsForStringFunction(absolute, argumentListTree), absolute);
+                    }
+                    case "toUpperCase": {
+                        return new ToUpperCase(getArgumentsForStringFunction(absolute, argumentListTree), absolute);
+                    }
+                    case "trim": {
+                        return new TrimString(getArgumentsForStringFunction(absolute, argumentListTree), absolute);
```

Comment:
Please add these new functions to 'Standalone Functions' section of 'record-path-guide.adoc'.